packages/quantic/force-app/main/default/lwc/quanticHeadlessLoader/quanticHeadlessLoader.js

```diff
@@ -305,7 +307,22 @@ async function initializeWithHeadless(element, engineId, initialize) {
     initialize(await getHeadlessEnginePromise(engineId));
     setComponentInitialized(element, engineId);
   } catch (error) {
-    console.error('Fatal error: unable to initialize component', error);
+    console.error(
+      `Fatal error: unable to initialize ${element?.template?.host?.localName} component.`,
+      error
+    );
+    element?.setInitializationError?.();
+    if (!componentErrorDisplayed) {
+      componentErrorDisplayed = true;
```

Comment:
@erocheleau I'm using this variable to make sure I'm displaying the Error dialog only once even if you have errors in multiple components.

For example if you have invalid properties in both the Quantic Pager and the Quantic Result Per Page. You will get only one Error Modal for whichever component threw first, let's say it's the Quantic Pager, the message in the modal will be about the Quantic Pager. When you close the modal, there is no other modal that appears. However in the page the Error component will be displayed in both the Quantic Pager and the Quantic Result Per Page.

After fixing the invalid properties of the Quantic Pager and reloading the page, the error modal will appear aagain but this time containing an error message about the Quantic Result Per Page component.

Reply:
Makes total sense, good idea